src/compiler/csharp_generator.cc

```diff
@@ -701,9 +709,17 @@ void GenerateBindServiceMethod(Printer* out, const ServiceDescriptor* service) {
   out->Indent();
   for (int i = 0; i < service->method_count(); i++) {
     const MethodDescriptor* method = service->method(i);
+    std::string method_name(method->name());
+    if (append_async_suffix == true) {
+      if (!method->client_streaming() && !method->server_streaming() && method_name.size() >= 5 &&
```

Comment:
I might be mis-reading -- shouldn't we use `Async` suffix on all four RPC types?